packages/shared/src/hooks/experiments/useSummaryActivation.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { ActionType } from '../../graphql/actions';
+import { useToastNotification } from '../useToastNotification';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useActions } from '../useActions';
+import { useConditionalFeature } from '../useConditionalFeature';
+import { feature } from '../../lib/featureManagement';
+
+interface UseSummaryActivationProps {
+  summary: string;
+}
+
+interface UseSummaryActivation {
+  onClickSummary: () => void;
+  shouldShowOverlay: boolean;
+  isLoading: boolean;
+}
+
+export const useSummaryActivation = ({
+  summary,
+}: UseSummaryActivationProps): UseSummaryActivation => {
+  const { displayToast } = useToastNotification();
+  const { user } = useAuthContext();
+  const timeoutRef = useRef<number>(null);
+  const { isActionsFetched, checkHasCompleted, completeAction } = useActions();
+  const shouldEvaluate =
+    !!user &&
+    !!summary &&
+    isActionsFetched &&
+    !checkHasCompleted(ActionType.ShowSummary);
+  const { value: isEnrolled } = useConditionalFeature({
+    shouldEvaluate,
+    feature: feature.generateSummary,
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const shouldShowOverlay = (isEnrolled && shouldEvaluate) || isLoading;
+
+  useEffect(() => {
+    return () => {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+    };
+  }, []);
+
+  const onClickSummary = useCallback(() => {
+    setIsLoading(true);
+    completeAction(ActionType.ShowSummary);
+
+    timeoutRef.current = window.setTimeout(() => {
```

Comment:
I don't think you need `window` here, `setTimeout` is fine as global hoisted function.

Reply:
Probably also clear previous timeout here if user clicks multiple times.

Reply:
Right. Sometimes the `setTimeout` from jest is getting imported automatically and I was going crazy as to why the app is getting broken. Just to ensure it is the actual function, I added `window`.

Reply:
Yeah the part about `window` is non blocking, I get that as well sometimes 🙏

Reply:
Added the cleanups now 🥳